lib/Doctrine/ODM/MongoDB/Aggregation/Stage/AbstractReplace.php

```diff
@@ -18,19 +18,14 @@
 
 abstract class AbstractReplace extends Operator
 {
-    /** @var string|mixed[]|Expr|null */
-    protected $expression;
     protected DocumentManager $dm;
-    protected ClassMetadata $class;
 
     /** @param string|mixed[]|Expr|null $expression */
-    final public function __construct(Builder $builder, DocumentManager $documentManager, ClassMetadata $class, $expression = null)
+    final public function __construct(Builder $builder, DocumentManager $documentManager, protected ClassMetadata $class, protected $expression = null)
```

Comment:
```suggestion
    final public function __construct(Builder $builder, DocumentManager $dm, protected ClassMetadata $class, protected $expression = null)
```
So the property could be promoted